### ISSUE TITLE
Design for  Backup Interface

### ISSUE THREAD

**Issue by shaase-ctrl** [kubermatic/dashboard#3456]
https://www.figma.com/file/dJQKgUPzIDND36o2Nq8dct/Loodse-UI?node-id=5247%3A36199

Please comment feedback:

**Comment by maciaszczykm** [comment]
I really like it. My remarks at the moment:

1. Can we change `Check settings` into something like `View settings`?
2. It seems that we will be keeping rather big amount of the automatic backups. I think that managing them from a table within a table may be problematic. For us as a developers but also for users. This is just a proposal but what do you think about adding specific details view? We could keep all the settings there visible all the time and have list of the backups as well.
3. Snapshots are just single-time backups?

**Comment by floreks** [comment]
Question regarding 1 point. Are those settings read-only or will use be able to update them?

2. +1

**Comment by maciaszczykm** [comment]
@floreks View only.

**Comment by floreks** [comment]
I also would prefer some other naming then. Simply `Settings` maybe?

**Comment by maciaszczykm** [comment]
SGTM.

**Comment by shaase-ctrl** [comment]
1. These are Backup Settings, the question is if this is redundant or needed?
2. I agree, but we might want to guide the user to not create 100 backup rules but instead promote reasonable usage.
3. Yes, snapshots are single-time backups.

**Comment by maciaszczykm** [comment]
@shaase-ctrl 2. Yep, but also single automatic backup might produce a lot of backups if we will pick short interval.

**Comment by cschieder** [comment]
1. Changed it to View settings
2. Made a draft for this https://www.figma.com/file/dJQKgUPzIDND36o2Nq8dct/Loodse-UI?node-id=5287%3A38543 please take a look

**Comment by maciaszczykm** [comment]
@cschieder Looks better in my opinion.

**Comment by cschieder** [comment]
## Backups Summary, Current State
### Needed functionalities
Automatic Backups
- Add Automatic Backups 
	- sub functionality: create custom interval
- Turn off automatic backups (stop automatic creation of backups, retain backups until expired)
- Delete backups (deletes backups, if all backups are deleted and the automatic backups are turned off, the automatic backups "group" gets "dissolved" and deleted as well) 
- "Force" Delete (turn off and delete backups at once, deletes entire automatic backups "group", maybe grouped into "Delete backups" dialog but optional/activateable? Otherwise just clears backups)
- As of right now we haven't planned if the backups generated by automatic backups should be renameable or if the user should be able to set a prefix or suffix that gets applied to every generated backup

**Notes:**
deleting backups/snapshots should have a warning and confirmation
a backup created by automatic backups has the same restoring options as a snapshot (Restore, Create Cluster)


### Snapshots
- Create Snapshot
- Delete Snapshot
- Restore
- Rename 
- Create Cluster (create new cluster from snapshot/backup)

**Open points**
- Are all needed functionalities covered and clear?
- How does "Destination" work? Is just a textfield fine as is?
- Should we use tabs or show both Automatic Backups and Snapshots on the same page?
- Do we have access to and should we show data like "availability zones" or "provider"? 
- Is every action final or do we offer an "undo" or a grace period in case of human-failure?

Basically anything that's open for debate or would prevent this design from being implemented, hopefully we can wrap my part of this up until the end of the week

**Comment by shaase-ctrl** [comment]
Very good summary. I think the functions are clear, we need to decide how we want to integrate the backup destinations as this is quite crucial. The common standard is an S3/compatible storage that every large CSP already has, but what about the vsphere / openstack / bare metal installations. We could go with an S3 endpoint for v1. Wdyt?

**Comment by shaase-ctrl** [comment]
I would like to the the Backup entry on the navigation again - in future we could provide more than cluster state backups and therefore and independent entry would make absolut sense.

**Comment by cschieder** [comment]
For now we are focusing on backups tied to projects and clusters, I believe it's the most intuitive for the user to have them in the place shown in my design. It makes sense to have the backups belonging to a project grouped under that project.

We don't have any concrete plans for other types of backups yet so I would be careful of trying to optimize for the future by sacrificing the UX of the present.

**Comment by maciaszczykm** [comment]
@cschieder @shaase-ctrl @kubermatic/sig-api @kubermatic/sig-ui Do you think if we should organize some meeting to push it forward?

**Comment by maciaszczykm** [comment]
Linking the feature docs: https://docs.kubermatic.com/kubermatic/v2.17/cheat_sheets/etcd/backup-and-restore/

**Comment by cschieder** [comment]
### Update from Backup Meeting 2021-07-02, 9:30
Should we use tabs or show both Automatic Backups and Snapshots on the same page? => **Going forward with tabs**
Is every action final or do we offer an "undo" or a grace period in case of human-failure? => **Undo feature to be discussed at a later date**

#### Notes and Ideas:
- Rename to Cluster State Backups
- Configure Destinations in a new tab, determine how

**Comment by maciaszczykm** [comment]
## Summary

### Current state
- Design is almost ready: https://www.figma.com/file/dJQKgUPzIDND36o2Nq8dct/Loodse-UI?node-id=5247%3A36199
- Backend is in experimental state: https://docs.kubermatic.com/kubermatic/v2.17/cheat_sheets/etcd/backup-and-restore/
- We need to start working on the API and the UI, some changes in the backend might be needed as well

### Core functionalities
- Users being able to create automatic cluster state backups that will create single-time backups in specified intervals
- Users being able to create single-time cluster state backups

### Other functionalities
- Users being able to manage destinations that can be later used in the backups (as a 3rd tab in the backups view)

### To do
- Test current functionality of the backups feature: https://docs.kubermatic.com/kubermatic/v2.17/cheat_sheets/etcd/backup-and-restore/
- Use the new backup controller in KKP by default (?) 
- Implement API to manage resources like `EtcdBackupConfig`, `EtcdRestore` etc. (required for UI)
- Implement UI

### Other comments
> By default a new backup configuration resource is created in the cluster namespace for each user cluster once the controllers are enabled. The new backup configuration resource is created with a default of @every 20m interval and 20 backup retention.

Perhaps we should put this info somewhere once the backup controllers will be enabled.

**Comment by maciaszczykm** [comment]
I think we can close design issue and start working on the implementation. In case of any comments please reopen it.
